File: ai-video-platform/frontend/src/types/credit.ts
export interface CreditBalance {
  credits: number;
  total_used: number;
}

export interface CreditTransaction {
  id: number;
  user_id: number;
  amount: number;
  balance_after: number;
  transaction_type: string;
  reference_type: string | null;
  reference_id: number | null;
  description: string | null;
  created_at: string | null;
}

export interface CreditPackage {
  id: number;
  name: string;
  credits: number;
  price_cents: number;
  is_active: boolean;
}

export interface CreditTransactionList {
  items: CreditTransaction[];
  total: number;
  page: number;
  per_page: number;
  total_pages: number;
}

File: ai-video-platform/backend/app/schemas/credit.py
from typing import Optional
import datetime
from pydantic import BaseModel


class CreditBalanceResponse(BaseModel):
    credits: int
    total_used: int


class CreditTransactionResponse(BaseModel):
    id: int
    user_id: int
    amount: int
    balance_after: int
    transaction_type: str
    reference_type: Optional[str] = None
    reference_id: Optional[int] = None
    description: Optional[str] = None
    created_at: Optional[datetime.datetime] = None

    model_config = {"from_attributes": True}


class CreditTransactionListResponse(BaseModel):
    items: list[CreditTransactionResponse]
    total: int
    page: int
    per_page: int
    total_pages: int


class CreditPackageResponse(BaseModel):
    id: int
    name: str
    credits: int
    price_cents: int
    is_active: bool

    model_config = {"from_attributes": True}


class CreditPurchaseRequest(BaseModel):
    package_id: int
